index.html

```diff
@@ -609,106 +609,211 @@
+    <p>
+      In order to push an NDEF record of URL type, simply use NFCMessage.
+    </p>
+    <pre class="highlight">
+navigator.nfc.push({
+    data: [{ recordType: "url", data: "/path/resource/" }];
+  }).then(() => {
+    console.log("Message pushed.");
+  }).catch((error) => {
+    console.log("Push failed :-( try again.");
+  });
+    </pre>
+  </aside>
 
-      // Now do some calculations and then update the state.
+  <aside title="Read and update data stored on tag" class="example">
+    <pre class="highlight">
+// Use default options for watch.
+navigator.nfc.watch((message) => {
+  console.log('NDEF message received from URL ' + message.url);
 
-      // Specify custom format, to control NDEF data layout.
-      var message = {
-        url: "https://company.com/path/game/update",  // app specific path
-        data: [{
-          recordType: "json",
-          mediaType: "application/json",
-          data: { level: 3, points: 4500, lives: 3 }
-        }]
-      };
+  if (message.data[0].recordType == 'empty') {
+    navigator.nfc.push({
+      url: "/mypath/myapp",  // path set by the application
+      data: [{
+        recordType: "url",
+        data: "Initial data"
+      }]
+    });
+  }
+  processMessage(message);
+}).then(() => {
+  console.log("Watch added.");
+}).catch((error) => {
+  console.log("Adding watch failed: " + error.name)
+});
+
+function processMessage(message) {
+  message.data.forEach((record) => {
+    if (record.recordType == "string") {
+      console.log('Data is string: ' + record.data);
+    } else if (record.recordType == 'json') {
+      processJSON(record.data);
+    } else if (record.recordType == 'url') {
+    console.log('Data is URL: ' + record.data);
+    } else if (record.recordType == 'opaque') {
+      processBinary(record.data);
+  });
+};
+
+function processBinary(data) {
+  console.log('Known (string) binary data: ');
+  console.log(String.fromCharCode.apply(null, new Uint16Array(data)));
+};
+
+function processJSON(data) {
+  var obj = JSON.parse(data);
+  console.log('JSON data: ' + obj.myProperty.toString());
+};
+    </pre>
+  </aside>
+
+  <aside title="Save and restore game progress with another device"
+       class="example">
+    <p>
+      Filtering of relevant data sources can be done by the use of
+      the <a>NFCWatchOptions</a>. Below we accept any URL with <code>
+      "/mypath/mygame/"</code> in its path. When we read the data,
+      we immediately update the game progress by issueing a push
+      with a custom NDEF data layout.
+    </p>
+    <p>
+      The example allows reading and pushing to both peers and tags,
+      whichever one is tapped first.
+    </p>
+    <pre class="highlight">
+navigator.nfc.watch(reader, { url: "*/mypath/mygame/*" });
+
+function reader(message) {
+  console.log("Source:     " + message.url);
+  console.log("Game state: " + message.data);
+
+  var message = {
+    url: "https://company.com/path/game/update",  // app specific path
```

Comment:
10.7.1.9 Creating a Web NFC Id
origin + NFCMessage.url , should be valid url.

If you don't like it, we can add more steps, like:

1. id = origin;
2. If NFCMessage.url has "origin" and origin of id and NFCMessage.url are same
  * Then id = NFCMessage.url
3. If NFCMessage.url doesn't have "origin", then id = id + NFCMessage.url
  * If id is valid url, WebNFC Id = id
  * else, SyntaxError